src/components/button-icon/button-icon.component.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import Events from '../../utils/helpers/events';
+import Icon from '../icon/icon.component';
+import StyledButtonIcon from './button-icon.style';
+
+const ButtonIcon = ({ onAction, type, ...rest }) => {
+  const onKeyDown = (e) => {
+    if (Events.isEnterKey(e) || Events.isSpaceKey(e) || Events.isEscKey(e)) {
+      e.preventDefault();
+      onAction(e);
+    } else {
+      e.stopPropagation();
+    }
+  };
+
+  const buttonRef = React.createRef();
+
+  const handleOnAction = (e) => {
+    e.preventDefault();
+    onAction(e);
+  };
```

Comment:
is there a reason for preventing default on click?

Reply:
not as far as I can see, added on autopilot without thinking it through, thanks!
fixed in: ea86715